src/connection.c

```diff
@@ -38,6 +39,8 @@ DECLARE_POOL(pool_head_connection,     "connection",     sizeof(struct connectio
 DECLARE_POOL(pool_head_conn_hash_node, "conn_hash_node", sizeof(struct conn_hash_node));
 DECLARE_POOL(pool_head_sockaddr,       "sockaddr",       sizeof(struct sockaddr_storage));
 DECLARE_POOL(pool_head_authority,      "authority",      PP2_AUTHORITY_MAX);
```

Comment:
Yes, this part of the design. This should avoid any unexpected increase in memory usage / performance penalties for existing HAProxy with authority TLVs. As these have a known size, using a dedicated pool should be more efficient. Therefore, I see no reason not to keep it.

Reply:
I agree that a dedicated pool is most efficient for authority TLVs. It might become an issue on memory pressure, since the memory used by the pool could be needed somewhere else. However, there is no easy fix for this.